### DISCUSSION TITLE
Compilation impossible for Tasmota since 14.0.0.0 with Tasmesh

### DISCUSSION THREAD

**Discussion by fdaubercy** [arendst/Tasmota#21395] [Support request] [answered]
Good morning.
I'm trying to compile Tasmota with Tasmesh on ESP32

However from firmware version 14.0.0.0, I get an error on platformio.
If I take version 13.4.0.0, I have no problem compiling with the following directives:
#define USE_TASMESH 
#define USE_TASMESH_HEARTBEAT 1 
#define TASMESH_OFFLINE_DELAY 3

I think it is due to the use of the Espressif32 platform version 2024.5.11

**Comment by Jason2866** [comment]
There is no v14 (yet) ;-)

**Reply by HofeBY** [reply]
A case of crystal ball fortune telling.

**Comment by fdaubercy** [comment]
With the pre-releases 14.0.0 or branch development.

**Reply by sfromis** [reply]
Pre-release is not for consumption. Use development or release when building.

**Comment by fdaubercy** [comment]
With realease v13.4.0, I compile tasmota without problem with the following parameters in the 'user_config_override.h' file:
#define USE_TASMESH
#define USE_TASMESH_HEARTBEAT 1
#define TASMESH_OFFLINE_DELAY 3

With the development branch, I get the following error message:
`
Compiling .pio\build\tasmota32-test\libd57\Adafruit_BusIO\Adafruit_BusIO_Register.cpp.o
C:/Users/fdaub/Documents/Github/Tasmota v13.4/Tasmota/tasmota/tasmota_xdrv_driver/xdrv_57_9_tasmesh.ino: In function 'void MESHstartBroker()':
C:/Users/fdaub/Documents/Github/Tasmota v13.4/Tasmota/tasmota/tasmota_xdrv_driver/xdrv_57_9_tasmesh.ino:435:27: error: no matches converting function 'CB_MESHDataReceived' to type 'esp_now_recv_cb_t' {aka 'void (*)(const struct esp_now_recv_info*, const unsigned char*, int)'}
  435 |   esp_now_register_recv_cb(CB_MESHDataReceived);
      |   ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
C:/Users/fdaub/Documents/Github/Tasmota v13.4/Tasmota/tasmota/tasmota.ino:3206:6: note: candidates are: 'void CB_MESHDataReceived(uint8_t*, uint8_t*, uint8_t)'
C:/Users/fdaub/Documents/Github/Tasmota v13.4/Tasmota/tasmota/tasmota_xdrv_driver/xdrv_57_9_tasmesh.ino:54:6: note:                 'void CB_MESHDataReceived(const uint8_t*, const uint8_t*, int)'
   54 | void CB_MESHDataReceived(const uint8_t *MAC, const uint8_t *packet, int len) {
      |      ^~~~~~~~~~~~~~~~~~~
Archiving .pio\build\tasmota32-test\libac0\libAdafruit_MAX31865-1.1.0-custom
`

**Comment by arendst** [comment]
ESP_Now has been redesigned with Core3 / IDF5

Tasmota currently does not support TasMesh on ESP32.

**Comment by arendst** [comment, marked as answer]
Try latest dev branch. I've changed the offending callback to use the core3 syntax but am unable to test it.

Pls give it a try and let me know if it fixes TasMesh in v13.4.1.2

**Comment by ferbulous** [comment]
Not related, but have you figured out how to retain the online status for tasmesh devices (LWA)?

**Comment by fdaubercy** [comment]
Hello everyone.
Thank you for the previous information.** I now have another problem.**

> For the broker:
- Wifi MAC address: 08:3A:F2:67:DC:30
- AP MAC address: 08:3A:F2:67:DC:31

> For the 1st test node:
- Wifi MAC address: B4:E6:2D:C9:D3:39
- AP MAC address: B4:E6:2D:C9:D3:3A

> I launch my Broker correctly with the following command :
- rule1 ON system#boot DO meshbroker ENDON
- rule2 ON mesh#broker>=1 DO publish cmnd/tasnodes/meshnode B4:E6:2D:C9:D3:3A ENDON
- Please note that the wifi connection password recorded for the broker and the node are the same

```
18:45:09.404 RUL: SYSTEM#BOOT performs "Backlog meshbroker;"
18:45:09.405 SRC: Rule
18:45:09.407 CMD: Grp 0, Cmd 'BACKLOG', Idx 1, Len 11, Pld -99, Data 'meshbroker;'
18:45:09.443 SRC: Backlog
18:45:09.445 CMD: Grp 0, Cmd 'MESHBROKER', Idx 1, Len 0, Pld -99, Data ''
18:45:09.453 MSH: Broker MAC 08:3A:F2:67:DC:31
18:45:09.471 RUL: MESH#BROKER>=1 performs "publish cmnd/tasnodes/meshnode B4:E6:2D:C9:D3:39"
18:45:09.472 SRC: Rule
18:45:09.474 CMD: Grp 0, Cmd 'PUBLISH', Idx 1, Len 40, Pld -99, Data 'cmnd/tasnodes/meshnode B4:E6:2D:C9:D3:39'
18:45:09.490 MQT: cmnd/tasnodes/meshnode = B4:E6:2D:C9:D3:39
18:45:09.492 MSH: set crypto key to PASSWORD1
18:45:09.494 MQT: stat/espnow/passerelle/bureau/RESULT = {"Publish":11}
18:45:09.948 SRC: MQTT
18:45:09.950 CMD: Grp 1, Cmd 'MESHNODE', Idx 1, Len 17, Pld -99, Data 'B4:E6:2D:C9:D3:39'
18:45:09.961 MQT: stat/espnow/passerelle/bureau/RESULT = {"Command":"Error"}
```

> I record the following rule in the 1st node:
- rule1 ON system#init DO Backlog MeshChannel 11; MeshNode 08:3A:F2:67:DC:31 ENDON (the broker's wifi channel is arbitrarily set to 11: the channel used for the wifi connection I presume)

**- Should the node remain connected to wifi?**
I don't see the launch of rule1 on the 1st node in its logs

> Here are my questions:
- Do I have to delete the wifi connection settings on the 1st node?
- Is rule 1 recorded in the 1st node wrong?
- For the meshnode command: what is the MAC address to use?
- Do I already have to register the node on the broker using the meshpeer command? If yes, which mac address to use?

Thank you all in advance for any help you can give me.

Frédéric
